opentnsim/core.py

```diff
@@ -482,50 +470,78 @@ def pass_edge(self, origin, destination):
 
             self.distance += distance
 
-            value = 0
-
-            # remember when we arrived at the edge
-            arrival = self.env.now
-
-            v = self.current_speed
-
-            # This is the case if we are sailing on power
-            if getattr(self, "P_tot_given", None) is not None:
-                edge = self.graph.edges[origin, destination]
-                depth = self.graph.get_edge_data(origin, destination)["Info"]["GeneralDepth"]
+        # remember when we arrived at the edge
+        arrival = self.env.now
+
+        v = self.current_speed
+
+        # This is the case if we are sailing on power
+        if getattr(self, "P_tot_given", None) is not None:
+            edge = self.graph.edges[origin, destination]
+            depth = self.graph.get_edge_data(origin, destination)["Info"]["GeneralDepth"]
+
+            # estimate 'grounding speed' as a useful upperbound
+            (
+                upperbound,
+                selected,
+                results_df,
+            ) = opentnsim.strategy.get_upperbound_for_power2v(self, width=150, depth=depth, margin=0)
+            v = self.power2v(self, edge, upperbound)
+            # use computed power
+            value = self.P_given
+
+        # Wait for edge resources to become available
+        # The logging expects a value, but there's not much to report, other than time and location.
+        # So we'll just use 0.
+        value = 0
+
+        if "Resources" in edge.keys():
+            resource = edge["Resources"]
+            request = resource.request()
+            self.log_entry_v0(
+                "Waiting to pass edge {} - {} start".format(origin, destination),
+                arrival,
+                value,
+                orig,
+            )
+            yield request
+            self.log_entry_v0(
+                "Waiting to pass edge {} - {} stop".format(origin, destination),
+                self.env.now,
+                value,
+                orig,
+            )
```

Comment:
Lines 498-513, the log info, the timestamp for: Waiting to pass edge {} - {} start and Waiting to pass edge {} - {} stop are exactly the same in all my simulation runs